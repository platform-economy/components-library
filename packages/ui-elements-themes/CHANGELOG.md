# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## [0.1.2](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.1...@relayr/ui-elements-themes@0.1.2) (2019-08-22)

**Note:** Version bump only for package @relayr/ui-elements-themes






## [0.1.1](https://github.com/relayr/ui-elements/compare/@relayr/ui-elements-themes@0.1.0...@relayr/ui-elements-themes@0.1.1) (2019-08-21)

**Note:** Version bump only for package @relayr/ui-elements-themes






# 0.1.0 (2019-08-02)


### Features

* **ui-elements-basic:** new InputLabel component ([abb6076](https://github.com/relayr/ui-elements/commit/abb6076))
* **ui-elements-layout:** add Card component ([e8db0cf](https://github.com/relayr/ui-elements/commit/e8db0cf))
* **ui-elements-tables:** Table components ([#102](https://github.com/relayr/ui-elements/issues/102)) ([b1b8d6f](https://github.com/relayr/ui-elements/commit/b1b8d6f))
* **ui-elements-themes:** custom font linking system ([8e9f0f0](https://github.com/relayr/ui-elements/commit/8e9f0f0))
