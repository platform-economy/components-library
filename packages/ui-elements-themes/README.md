# ui-elements-themes

## Add to project

via npm:

```bash
npm install --save @relayr/ui-elements-themes
```

or Yarn:
```bash
yarn add @relayr/ui-elements-themes
```
