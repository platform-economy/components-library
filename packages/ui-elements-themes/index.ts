export * from './themes/light';
export * from './themes/dark';
export * from './themes/test';
export * from './themes/_common/paletteDark';
export * from './themes/_common/paletteLight';
export * from './themes/_common/methods';
export * from './themes/_common/fonts';
export * from './themes/_common/withTestTheme';
