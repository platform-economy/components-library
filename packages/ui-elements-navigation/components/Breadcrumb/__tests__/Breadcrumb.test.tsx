import * as React from 'react';
import * as TestRenderer from 'react-test-renderer';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

/*
A set of utilities for testing Styled Components with Jest.
This package improves the snapshot testing experience and
provides a brand new matcher to make expectations on the style rules.
*/
import 'jest-styled-components';

// Component
import { Breadcrumb } from '../index';
import { withTestTheme } from '@relayr/ui-elements-themes';

enzyme.configure({ adapter: new Adapter() });

describe('ui-elements-navigation', () => {
  describe('Components', () => {
    describe('Breadcrumb', () => {

      it('should render with default state', () => {
        const tree = TestRenderer
          .create(
            withTestTheme(
              <Breadcrumb/>,
            ),
          )
          .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should match snapshot', () => {
        const tree = TestRenderer
        .create(
          withTestTheme(
            <Breadcrumb>
              <Breadcrumb.Item>
                <button>Test</button>
              </Breadcrumb.Item>
              <Breadcrumb.Item>
                <button>Test2</button>
              </Breadcrumb.Item>
            </Breadcrumb>,
          ),
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
      });

      it('should handle isCurrent props', () => {
        const tree = TestRenderer
        .create(
          withTestTheme(
            <Breadcrumb>
              <Breadcrumb.Item>
                <button>Test</button>
              </Breadcrumb.Item>
              <Breadcrumb.Item isCurrent={true}>
                <button>Test2</button>
              </Breadcrumb.Item>
            </Breadcrumb>,
          ),
        )
        .toJSON();
        expect(tree).toMatchSnapshot();
      });

    });
  });
});
