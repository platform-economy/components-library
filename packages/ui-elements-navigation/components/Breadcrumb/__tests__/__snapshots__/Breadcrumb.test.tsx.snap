// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui-elements-navigation Components Breadcrumb should handle isCurrent props 1`] = `
.c1 {
  display: inline;
  padding: 10px;
  cursor: default;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 {
  display: inline;
  padding: 10px;
  cursor: default;
  font-weight: bold;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  width: auto;
  height: 1em;
  font-size: 1em;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: -0.48px;
  -moz-letter-spacing: -0.48px;
  -ms-letter-spacing: -0.48px;
  letter-spacing: -0.48px;
  color: #text;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    <button>
      Test
    </button>
  </span>
  <span
    className="c2"
  >
    <button>
      Test2
    </button>
  </span>
</div>
`;

exports[`ui-elements-navigation Components Breadcrumb should match snapshot 1`] = `
.c1 {
  display: inline;
  padding: 10px;
  cursor: default;
  font-weight: normal;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  width: auto;
  height: 1em;
  font-size: 1em;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: -0.48px;
  -moz-letter-spacing: -0.48px;
  -ms-letter-spacing: -0.48px;
  letter-spacing: -0.48px;
  color: #text;
}

<div
  className="c0"
>
  <span
    className="c1"
  >
    <button>
      Test
    </button>
  </span>
  <span
    className="c1"
  >
    <button>
      Test2
    </button>
  </span>
</div>
`;

exports[`ui-elements-navigation Components Breadcrumb should render with default state 1`] = `
.c0 {
  width: auto;
  height: 1em;
  font-size: 1em;
  font-weight: normal;
  font-style: normal;
  font-stretch: normal;
  line-height: normal;
  -webkit-letter-spacing: -0.48px;
  -moz-letter-spacing: -0.48px;
  -ms-letter-spacing: -0.48px;
  letter-spacing: -0.48px;
  color: #text;
}

<div
  className="c0"
/>
`;
